idseq_dag/steps/run_srst2.py

```diff
@@ -0,0 +1,113 @@
+import os
+import pandas as pd
+
+from idseq_dag.engine.pipeline_step import PipelineStep
+from idseq_dag.util.s3 import fetch_from_s3
+import idseq_dag.util.command as command
+
+class PipelineStepRunSRST2(PipelineStep):
+    '''
+    srst2 is used to detect resistance genes.
+    See: https://github.com/katholt/srst2
+    '''
+    def run(self):
+        ''' Invoking srst2 '''
+        db_file_path = fetch_from_s3(self.additional_files["resist_gene_db"], self.output_dir_local)
+        min_cov = str(self.additional_attributes['min_cov'])
+        n_threads = str(self.additional_attributes['n_threads'])
+        if len(self.input_files_local[0]) == 2:
+            fwd_rd = self.input_files_local[0][0]
+            rev_rd = self.input_files_local[0][1]
+            # Rename to format srst2 expects
+            # TODO: Handle Fasta files with --read_type f option
+            command.execute(f"ln -sf {fwd_rd} _R1_001.fastq.gz")
+            command.execute(f"ln -sf {rev_rd} _R2_001.fastq.gz")
+            srst2_params = [
+                'srst2', '--input_pe', '_R1_001.fastq.gz', '_R2_001.fastq.gz', '--forward', '_R1_001', '--reverse', '_R2_001',
+                '--min_coverage', min_cov,'--threads', n_threads, '--output', os.path.join(self.output_dir_local, 'output'),
+                '--log', '--gene_db', db_file_path
+            ]
+        else:
+            rd = self.input_files_local[0][0]
+            command.execute(f"ln -sf {rd} rd.fastq.gz")
+            srst2_params = [
+                'srst2', '--input_se', 'rd.fastq.gz', '--min_coverage', min_cov,'--threads', n_threads,
+                '--output', self.output_dir_local+'/output', '--log', '--gene_db', db_file_path
+            ]
+        command.execute(" ".join(srst2_params))
+        log = os.path.join(self.output_dir_local, 'output.log')
+        log_dest = self.output_files_local()[0]
+        results = os.path.join(self.output_dir_local, 'output__genes__ARGannot_r2__results.txt')
+        results_dest = self.output_files_local()[1]
+        os.rename(log, log_dest)
+        os.rename(results, results_dest) 
+        if not os.path.exists(os.path.join(self.output_dir_local, 'output__fullgenes__ARGannot_r2__results.txt')): 
+            PipelineStepRunSRST2.fill_file_path(self.output_files_local()[2])
+            PipelineStepRunSRST2.fill_file_path(self.output_files_local()[3])
+            PipelineStepRunSRST2.fill_file_path(self.output_files_local()[4])
+        else:
+            # Post processing of amr data
+            results_full = os.path.join(self.output_dir_local, 'output__fullgenes__ARGannot_r2__results.txt')
+            results_full_dest = self.output_files_local()[2]
+            PipelineStepRunSRST2.mv_to_dest(results_full, results_full_dest)
```

Comment:
what is this mv_to_dest for? why not just do move?

Reply:
@yunfangjuan -- It was a helper function to deal with the case where `mv` errors because source and destination are the same. I recommended `os.path.rename` because it handles that case properly. Do you know a better python function to use? Maybe `shutil.move` is the answer after all.

Reply:
@yunfangjuan @cdebourcy yes, maybe shutil.move is better? would avoid the need for handling mv errors, it looks like.  
"shutil.move(src, dst)
Recursively move a file or directory (src) to another location (dst).

If the destination is an existing directory, then src is moved inside that directory. If the destination already exists but is not a directory, it may be overwritten depending on os.rename() semantics.

If the destination is on the current filesystem, then os.rename() is used. Otherwise, src is copied (using shutil.copy2()) to dst and then removed."